nomenclature/processor/region.py

```diff
@@ -361,21 +365,32 @@ def apply(self, df: IamDataFrame, dsd: DataStructureDefinition) -> IamDataFrame:
                             # Second, special weighted aggregation
                             for var, kwargs in vars_kwargs.items():
                                 if "region-aggregation" not in kwargs:
-                                    processed_dfs.append(
-                                        model_df.aggregate_region(
-                                            var, *regions, **kwargs
-                                        )
+                                    _df = _aggregate_region(
+                                        model_df,
+                                        var,
+                                        *regions,
+                                        **kwargs,
                                     )
+                                    if _df is not None and not _df.empty:
+                                        processed_dfs.append(_df)
                                 else:
                                     for rename_var in kwargs["region-aggregation"]:
                                         for _rename, _kwargs in rename_var.items():
-                                            processed_dfs.append(
-                                                model_df.aggregate_region(
-                                                    var, *regions, **_kwargs
-                                                ).rename(variable={var: _rename})
+                                            _df = _aggregate_region(
+                                                model_df,
+                                                var,
+                                                *regions,
+                                                **_kwargs,
                                             )
+                                            if _df is not None and not _df.empty:
+                                                processed_dfs.append(
+                                                    _df.rename(variable={var: _rename})
+                                                )
 
-        return pyam.concat(processed_dfs)
+        if processed_dfs:
+            return pyam.concat(processed_dfs)
+        else:
+            return IamDataFrame(pd.DataFrame([], columns=df.dimensions + ["value"]))
```

Comment:
If none of the region processing worked I would throw an error as that might lead to confusions along the lines of, "The region processing went through without any errors but I don't see any data".

```suggestion
            raise ValueError(The resulting data frame is empty please check the logs as to why that might be the case.)
```
Or something along those lines

Reply:
Good suggestion, implemented - but without reference to the log because I don't think anything meaningful is written to the log in most cases.

Reply:
Pefect, thanks.